src/libraries/System.Text.RegularExpressions/tests/Regex.KnownPattern.Tests.cs

```diff
@@ -137,13 +142,759 @@ string DomainMapper(Match match)
+            {
+                if (match.Groups[1].Success)
+                {
+                    privateDocument += match.Groups[1].Value;
+                }
+                else
+                {
+                    publicDocument += match.Groups[3].Value;
+                    privateDocument += match.Groups[3].Value;
+                }
+            }
+
+            Assert.Equal("This is not for public consumption.\rBut this is for public consumption.\rAgain, this is confidential.\n", privateDocument);
+            Assert.Equal("But this is for public consumption.\r", publicDocument);
+        }
+
+        // https://docs.microsoft.com/en-us/dotnet/standard/base-types/details-of-regular-expression-behavior#net-framework-engine-capabilities
+        [Theory]
+        [InlineData(RegexOptions.None)]
+        [InlineData(RegexOptions.Compiled)]
+        public void Docs_EngineCapabilities_RightToLeftMatching(RegexOptions options)
+        {
+            const string GreedyPattern = @".+(\d+)\.";
+            const string Input = "This sentence ends with the number 107325.";
+
+            // Match from left-to-right using lazy quantifier .+?.
+            Match match = Regex.Match(Input, GreedyPattern);
+            Assert.True(match.Success);
+            Assert.Equal("5", match.Groups[1].Value);
+
+            // Match from right-to-left using greedy quantifier .+.
+            match = Regex.Match(Input, GreedyPattern, RegexOptions.RightToLeft | options);
+            Assert.True(match.Success);
+            Assert.Equal("107325", match.Groups[1].Value);
+        }
+
+        // https://docs.microsoft.com/en-us/dotnet/standard/base-types/details-of-regular-expression-behavior#net-framework-engine-capabilities
+        [Theory]
+        [InlineData(RegexOptions.None)]
+        [InlineData(RegexOptions.Compiled)]
+        public void Docs_EngineCapabilities_PositiveNegativeLookbehind(RegexOptions options)
+        {
+            const string Pattern = @"^[A-Z0-9]([-!#$%&'.*+/=?^`{}|~\w])*(?<=[A-Z0-9])$";
+
+            Assert.True(Regex.IsMatch("jack.sprat", Pattern, RegexOptions.IgnoreCase | options));
+            Assert.False(Regex.IsMatch("dog#", Pattern, RegexOptions.IgnoreCase | options));
+            Assert.True(Regex.IsMatch("dog#1", Pattern, RegexOptions.IgnoreCase | options));
+            Assert.True(Regex.IsMatch("me.myself", Pattern, RegexOptions.IgnoreCase | options));
+            Assert.False(Regex.IsMatch("me.myself!", Pattern, RegexOptions.IgnoreCase | options));
+        }
+
+        // https://docs.microsoft.com/en-us/dotnet/standard/base-types/miscellaneous-constructs-in-regular-expressions#inline-options
+        [Theory]
+        [InlineData(RegexOptions.None)]
+        [InlineData(RegexOptions.Compiled)]
+        public void Docs_InlineOptions(RegexOptions options)
+        {
+            const string Input = "double dare double Double a Drooling dog The Dreaded Deep";
```

Comment:
Some of these are pretty wacky!!

Reply:
Yeah.  I refrained from applying any editorial oversight :)